### DISCUSSION TITLE
How to make the scrollbar of the datagrid always visible

### DISCUSSION THREAD

**Discussion by pyking-wang** [AvaloniaUI/Avalonia#17352] [Q&A] [answered]
```<DataGrid ItemsSource="{Binding CloudList}"  Height="200" IsReadOnly="True" ScrollViewer.AllowAutoHide="False"  ScrollViewer.VerticalScrollBarVisibility="Visible"/>```
```<DataGrid ItemsSource="{Binding CloudList}"  Height="200" IsReadOnly="True"  VerticalScrollBarVisibility="Visible"/>```

I am unable to set it up using these two methods, and ’visible‘ and ’auto‘ have the same effect.
ps:avalonia_version:11.1.1,and the control is in usercontrol

**Comment by 22lbo** [comment]
Since this question is still unanswered i thought i might add my solution, in case it helps anyone else.

This seems to work for me. 
```
    <ScrollViewer HorizontalScrollBarVisibility="Visible">
        <DataGrid ItemsSource="{Binding AllAssets}" GridLinesVisibility="All" AutoGenerateColumns="True" IsReadOnly="True">
        </DataGrid>
    </ScrollViewer>
```
I put the DataGrid inside the ScrollViewer like shown [here](https://github.com/AvaloniaUI/Avalonia/discussions/7458#discussioncomment-2068955) and then specify with `HorizontalScrollBarVisibility="Visible"` that I want the horizontal scroll bar to always be visible, which is what I need for my specific case. This simply makes it so that I dont have to scroll to the bottom of my 1000+ items list before I can see the horizontal scrollbar.

**Reply by pyking-wang** [reply, marked as answer]
Thank you for your reply. This is what I did and it solved my problem:
```
<Style Selector="DataGrid ScrollBar:vertical">
	<Setter Property="AllowAutoHide" Value="False"/>
</Style>
```
**horizontal** is the same.

**Comment by Nick-Crane** [comment]
For anyone struggling with this on version 11 the marked answer no longer works.
The solution is to set the AllowAutoHide property of the attached ScrollViewer of the DataGrid.
`<DataGrid ItemsSource="{Binding DataTable}" ScrollViewer.AllowAutoHide="False">
...
</DataGrid>`

Edit: Actually, the solution might depend on the Theme being used. The above works for the Fluent Theme.

**Reply by pyking-wang** [reply]
Yes. After upgrading to Avalonia 11.3.12 or later, use the following style instead:
`<Style Selector="DataGrid">
    <Setter Property="ScrollViewer.VerticalScrollBarVisibility" Value="Visible" />
    <Setter Property="ScrollViewer.AllowAutoHide" Value="False" />
</Style>`
